crates/ott-common/src/discovery/dns.rs

```diff
@@ -63,3 +63,22 @@ impl ServiceDiscoverer for DnsServiceDiscoverer {
         DiscoveryMode::Polling(Duration::from_secs(10))
     }
 }
+
+mod test {
+    #[tokio::test]
+    async fn server_deserializes_correctly() {
+        let dns_discovery_config = DnsDiscoveryConfig {
+            service_port: 8080,
+            dns_server: Option::Some(SocketAddr::new(
+                IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)),
+                8080,
+            )),
+            query: "".to_string(),
+        };
```

Comment:
This is just constructing the struct. Deserializing is the process of going from a format like `json` or `toml` into a native data type, like `DnsDiscoveryConfig`.